File: Excel/700-799/775/775 Challenge.py
import pandas as pd

path = "700-799/775/775 Pivot.xlsx"
input = pd.read_excel(path, usecols="A:B", skiprows=1, nrows=13)
test  = pd.read_excel(path, usecols="D:F", skiprows=1, nrows=6)

input['group'] = (input['Value1'] == "Org").cumsum()
input['row'] = input.groupby('group').cumcount() + 1
result = input.pivot(index=['group','row'], columns='Value1', values='Value2').reset_index()
result['Org'] = result['Org'].ffill()
result['Product'] = result.groupby('Org')['Product'].ffill()
result[['Version','Product']] = result.groupby('Org')[['Version','Product']].bfill()
result = result.drop(columns=['group', 'row'])
result = result.drop_duplicates().reset_index(drop=True)

print(result.equals(test)) # True

File: Excel/700-799/775/775 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/775/775 Pivot.xlsx"
input = read_excel(path, range = "A2:B15")
test  = read_excel(path, range = "D2:F8")

result = input %>%
  mutate(group = cumsum(Value1 == "Org")) %>%
  group_by(group) %>%
  mutate(row = row_number()) %>%
  pivot_wider(names_from = Value1, values_from = Value2) %>%
  fill(Org, Product, .direction = "down") %>%
  fill(Product, Version, .direction = "up") %>%
  ungroup() %>%
  select(-group, -row) %>%
  distinct()

all.equal(result, test, check.attributes = FALSE)
# > [1] TRUE